cluster_kwargs.yaml

```diff
@@ -54,6 +54,13 @@ uber_lyft:
   n_workers: 20
   worker_vm_types: [m6i.xlarge] # 4 CPU, 16 GiB (preferred default instance)
 
+# For tests/workflows/test_pytorch_optuna.py
+pytorch_optuna:
+  n_workers: 4
+  worker_gpu: 1
+  worker_options:
+    nthreads: 1
```

Comment:
Why is `nthreads=1` needed here?

Reply:
To prevent any concurrent access to the GPU per VM, since we're using single GPU instances here... could be possible to modify if the VMs had multiple devices I think.

Reply:
Do you want single worker per GPU? Or do you want single worker per host VM, and PyTorch will manage across all GPUs on the host? Or not sure?

Reply:
Is this based on a recommendation from our docs?

* If so, all good
* If not and this is indeed how one has to set it up, we should document this

Reply:
It appears having more than 1 thread per worker creates some sort of contention for the GPU and the utilization drops to ~40%, otherwise w/ 1 thread per worker I get nearly 100%. 

I did not get this from any documentation, nor did I locate it in the docs now; so will be sure to add a note about this experience in a follow-up PR there.

Reply:
Nvm, after going to make a PR to update docs, I saw it was (kinda) there after-all. Granted it doesn't explicitly say 1 thread per worker, but makes the overall point. https://docs.dask.org/en/latest/gpu.html?highlight=gpu#restricting-work

Reply:
I believe `dask-cuda` also does that for you. It might make more sense to try that rather than doing this manually.

Reply:
Okay, so you propose these lines should be removed, add `dask-cuda` then use the settings described here: https://docs.rapids.ai/api/dask-cuda/nightly/api/#client-initialization

As it appears the other configuration for it is to replace `LocalCluster` with `LocalCUDACluster` which doesn't seem like what we want(?).

Reply:
Add `dask-cuda` and then set worker class when making cluster, as shown in https://docs.coiled.io/user_guide/gpu.html

Reply:
Thanks. https://github.com/coiled/benchmarks/pull/787/commits/be11248966d5ec684562a3d5ae2f152a71b72c88

Reply:
I had tried this earlier, and ran into conda solve issues.  Bringing in the RAPIDS channel was unpleasant.  If it works now then grand.  I also think that this is a "I wanted a banana, but I got a gorilla holding a banana" situation.

Reply:
Ah, indeed. [Seems that's true here then. ](https://github.com/coiled/benchmarks/actions/runs/4872308102?pr=787)

Reply:
Thanks for the context all. Sounds like we should just stick with `nthreads=1` for now. I added a small comment here that mentions why we're making things single-threaded (linking back to this thread on GitHub).